envoy/http/header_validator.h

```diff
@@ -81,9 +81,21 @@ class HeaderValidator {
    * Returning the Reject value causes HTTP requests to be rejected with the 502 status,
    * and gRPC requests with the the UNAVAILABLE (14) error code.
    */
-  using ResponseHeaderMapValidationResult = RejectResult;
-  virtual ResponseHeaderMapValidationResult
-  validateResponseHeaderMap(ResponseHeaderMap& header_map) PURE;
+  virtual ValidationResult validateResponseHeaderMap(ResponseHeaderMap& header_map) PURE;
+
+  /**
+   * Validate the entire request trailer map.
+   * Returning the Reject value causes HTTP requests to be rejected with the 502 status,
+   * and gRPC requests with the the UNAVAILABLE (14) error code.
```

Comment:
should the other functions also include gRPC resposnes or is it trailers specific?

Reply:
All other functions include gRPC responses if they are possible. Response trailers error can only result on reset.